Strict version

davidpdrsn opened the pull request:
Fixes #1

Adds `assert_json_exact_eq!` which panics unless the two JSON values are exactly identical. It prints similar looking error messages.

Example usage:

```rust
assert_json_exact_eq!(
    json!({ "a": { "b": true } }),
    json!({ "a": {} })
);
```

This would panic with the message:

```
json atom at path ".a.b" is missing from rhs
```

And if the key was there but the values didn't match it would print

```
json atoms at path ".a.b" are not equal:
    lhs:
        true
    rhs:
        false
```

So basically the same output except it uses `lhs` and `rhs` rather than `expected` and `actual`.

I also decided to rename `assert_json_eq!` to `assert_json_include!` to better communicate the fact that it isn't an exact match. I'm open to other naming suggestions. This is a breaking change, but since the API was never really nailed down I'm fine with that.

---

## TODO
- [ ] Update readme
- [ ] Crate level docs

rofrol commented:
`assert_json_exact_eq` seems redundant. I would rather do:

`assert_json_eq` -> `assert_json_include`

`assert_json_exact_eq` -> `assert_json_eq`

davidpdrsn commented:
@rofrol I agree. I went with that naming because I didn't want to introduce breaking changes, but I guess that was a bit silly.

I've renamed the macros to `assert_json_eq` and `assert_json_include`.

Also updated the docs.

rofrol commented:
great
